child-processes/index.md

```diff
@@ -1,8 +1,189 @@
+
+Using multiple processes allows a Node application to scale. Node is designed for building distributed applications with many nodes, hence it's name Node.
+
+Pipes for `stdin`, `stdout`, and `stderr` are established between the parent Node process and the spawned subprocess. The behaviour matches that of pipes in the shell.
+
+The examples in this article are all Linux-based. On Windows, you will need to switch the commands I use with their Windows alternatives or use a Linux-like shell.
+
+# `exec` vs `spawn`
+
+There are two main approaches to running child processesL `exec` and `spawn`.
+
+| `exec`                                                     | `spawn`                                                     |
+|------------------------------------------------------------|-------------------------------------------------------------|
+| spawns a shell in which the command is executed            | Does not spawn a shell                                      |
+| buffers the data (waits until the process finishes and transfers all the data ) | Streams the data returned by the child process (data flow is constant) |
+| maximum data transfer 200kb (by default                    | has no data transfer size limit                             |
+
+Typically, `spawn` is more suitable for long-running process with large outputs. spawn streams input/output with child process. `exec` buffered output in a small (by default 200K) buffer.
+
+## A Simple `exec`
+
+In this example, we will call the `ls` command to listwith the optional param `-l` to show a long list of details. 
+<div class="repl-code">
+
+```javascript
+
+const { exec } = require('child_process');
+
+exec('ls -l', (err, stdout, stderr) => {
+  console.log(stdout);
+});
+
+```
+</div>
+
+For more details of the options that can be passed to `exec`, please see https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback
+
+# spawn
+
+The `spawn` method works by returning a new process. The process object that is returned has properties for each standard IO represented as a `Stream`: `.stdin` (`WriteStream`), `.stout` (`ReadStream`) and `.stderr` (`ReadStream`).
+ 
+We will now implement the previous example using the `spawn` method:
+
+<div class="repl-code">
+
+```javascript
+const { spawn } = require('child_process');
+const ls = spawn('ls', ['-l']);
+
+ls.stdout.on('data', (data) => {
+  console.log(`stdout: ${data}`);
+});
+
+ls.stderr.on('data', (data) => {
+  console.error(`stderr: ${data}`);
+});
+
+ls.on('close', (code) => {
+  console.log(`child process exited with code ${code}`);
+});
+
+```
+</div>
+
+In this example we are listening to the `stout` and `stderr` streams for `data` events, as well as listening for a `close` event.
```

Comment:
Are these events different from the exec approach?

Reply:
Yes - exec buffers it to strints

Reply:
Strings.

I'll add that :)